src/SelfService/Infrastructure/Persistence/GenericRepository.cs

```diff
@@ -7,44 +8,58 @@ namespace SelfService.Infrastructure.Persistence;
 public class GenericRepository<T, TId> : IGenericRepository<T, TId>
     where T : Entity<TId>
 {
-    private readonly DbSet<T> _dbSetReference;
+    protected readonly DbSet<T> DbSetReference;
 
     public GenericRepository(DbSet<T> dbSetReference)
     {
-        _dbSetReference = dbSetReference;
+        DbSetReference = dbSetReference;
     }
 
     public async Task Add(T model)
     {
-        await _dbSetReference.AddAsync(model);
+        await DbSetReference.AddAsync(model);
     }
 
     public async Task<bool> Exists(TId id)
     {
-        var found = await _dbSetReference.FindAsync(id);
+        var found = await DbSetReference.FindAsync(id);
         return found != null;
     }
 
-    public async Task<T?> FindBy(TId id)
+    public async Task<T?> FindById(TId id)
     {
-        return await _dbSetReference.FindAsync(id);
+        return await DbSetReference.FindAsync(id);
+    }
+
+    public async Task<T?> FindByPredicate(Func<T, bool> predicate)
+    {
+        // Done in two steps to avoid Entity Framework Core from choking on the predicate.
```

Comment:
My gut feeling says that this would be an issue if our data sets were bigger.
As is, this seems like a good workaround